rotkehlchen/api/rest.py

```diff
@@ -1077,17 +1077,27 @@ def get_asset_movements(
 
         return {'result': result, 'message': msg, 'status_code': status_code}
 
-    def add_history_event(self, event: 'HistoryBaseEntry') -> Response:
+    def add_history_event(self, main_event: 'HistoryBaseEntry', related_events: list['HistoryBaseEntry'] | None) -> Response:  # noqa: E501
         db = DBHistoryEvents(self.rotkehlchen.data.db)
         with self.rotkehlchen.data.db.user_write() as cursor:
             try:
                 identifier = db.add_history_event(
                     write_cursor=cursor,
-                    event=event,
+                    event=main_event,
                     mapping_values={
                         HISTORY_MAPPING_KEY_STATE: HISTORY_MAPPING_STATE_CUSTOMIZED,
                     },
                 )
+
+                if related_events is not None and len(related_events) > 0:
+                    for event in related_events:
+                        db.add_history_event(
+                            write_cursor=cursor,
+                            event=event,
+                            mapping_values={
```

Comment:
can you do it in a transaction? If you for example two events you don't want one being added and the other missing in the db. If the transaction fails you want to revert any already added event